### ISSUE TITLE
leading underscore results in handlebars template not getting generated

### ISSUE THREAD

**Issue by kappasims** [madskristensen/WebEssentials2013#1723]
"_nav.hbs" will not generate an associated .hbs.js file, but "nav.hbs" will on save or when selecting the "Compile Handlebars Template" command. This appears to be by design, but Ember.js partials require a leading underscore.

**Comment by am11** [comment]
WE considers `_filename.{insert-superset-lang-ext-here}` to be a partial file and hence does not generate the output for it.

Having said that, if the chained compilation is turned on for that language (in Tools > Options > Web Essentials), then saving a partial does case compilation of the caller(s) of the partial under question.

This is by design with the assumption that it should work for all scenarios. Does Ember require output for the partial file?

**Comment by kappasims** [comment]
Yes, Ember's partials are handlebars files that get compiled to javascript and added to an in-memory collection where the key they are referenced by is the name of the file. This can obviously be worked around by handling the compilation in a separate build task but if compiled templates generated by WE are to be used in the app, it doesn't make much sense to only be able to reference non-partial templates.

**Comment by kappasims** [comment]
With that said, using the WE-generated files still does require a manual wiring-up to add the compiled objects to the Ember application, so it may or may not be worth it.

**Comment by am11** [comment]
AFAIR, at some point we were thinking about revoking that hard-coded condition for partials and adding it to default `.weignore`. The stoppers were: 1) we weren't aware of any use-case where partial compilation is required, 2) `.weignore` require unit tests and some polishing before it can be used for this kind of functionality.
